mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/transformation/mpsToModelix/initial/ModuleSynchronizer.kt

```diff
@@ -135,64 +155,75 @@ class ModuleSynchronizer(private val branch: IBranch) {
+                    BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.reexport,
+                    dependency.isReexport.toString(),
+                )
+
+                cloudDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.uuid,
+                    targetModuleId,
+                )
+
+                cloudDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.name,
+                    moduleReference.moduleName,
+                )
+
+                val moduleId = moduleReference.moduleId
+                val isExplicit = if (module is Solution) {
+                    module.moduleDescriptor.dependencies.any { it.moduleRef.moduleId == moduleId }
+                } else {
+                    module.declaredDependencies.any { it.targetModule.moduleId == moduleId }
+                }
+                cloudDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.explicit,
+                    isExplicit.toString(),
+                )
+
+                val version = (module as? Solution)?.let {
+                    it.moduleDescriptor.dependencyVersions.filter { dependencyVersion -> dependencyVersion.key == moduleReference }
+                        .firstOrNull()?.value
+                } ?: 0
+                cloudDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.version,
+                    version.toString(),
+                )
+
+                cloudDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.scope,
+                    dependency.scope.toString(),
+                )
             }
-            cloudDependency.setPropertyValue(
-                BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.explicit,
-                isExplicit.toString(),
-            )
-
-            val version = (module as? Solution)?.let {
-                it.moduleDescriptor.dependencyVersions.filter { dependencyVersion -> dependencyVersion.key == moduleReference }
-                    .firstOrNull()?.value
-            } ?: 0
-            cloudDependency.setPropertyValue(
-                BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.version,
-                version.toString(),
-            )
-
-            cloudDependency.setPropertyValue(
-                BuiltinLanguages.MPSRepositoryConcepts.ModuleDependency.scope,
-                dependency.scope.toString(),
-            )
 
             dependencyBindings
         }
 
     private fun synchronizeModuleProperties(cloudModule: INode, module: SModule) {
         cloudModule.setPropertyValue(
             BuiltinLanguages.MPSRepositoryConcepts.Module.id,
+            // if you change this property here, please also change above where we check if the module already exists in its parent node
```

Comment:
Create a reusable function that extracts the unique identifier from `module`.

Reply:
fixed in c56e31430575b882f83bb97aad62953e88f1072d